modules/contact-form/js/gutenblock.jsx

```diff
@@ -0,0 +1,630 @@
+	Component,
+	Fragment
+} from '@wordpress/element';
+
+import {
+	__
+} from '@wordpress/i18n';
+
+/**
+ * Components:
+ */
+
+class GrunionForm extends Component {
+	constructor( ...args ) {
+		super( ...args );
+		this.onChangeSubject = this.onChangeSubject.bind( this );
+		this.onChangeTo = this.onChangeTo.bind( this );
+		this.onChangeSubmit = this.onChangeSubmit.bind( this );
+	}
+
+	onChangeSubject( x ) {
+		this.props.setAttributes( { subject: x } );
+	}
+
+	onChangeTo( x ) {
+		this.props.setAttributes( { to: x } );
+	}
+
+	onChangeSubmit( x ) {
+		this.props.setAttributes( { submit_button_text: x } );
+	}
+
+	render() {
+		return (
+			<Fragment>
+				<InspectorControls>
+					<PanelBody title={ __( 'Submission Details', 'jetpack' ) }>
+						<TextControl
+							label={ __( 'What would you like the subject of the email to be?' ) }
+							value={ this.props.subject }
+							onChange={ this.onChangeSubject }
+						/>
+						<TextControl
+							label={ __( 'Which email address should we send the submissions to?' ) }
+							value={ this.props.to }
+							onChange={ this.onChangeTo }
+						/>
+						<TextControl
+							label={ __( 'What should the label on the form’s submit button say?' ) }
+							value={ this.props.submit_button_text }
+							placeholder={ __( 'Submit »' ) }
+							onChange={ this.onChangeSubmit }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				<div className="grunion-form">
+					{this.props.children}
+					<TextControl
+						className="button button-primary button-default grunion-submit-button"
+						value={ this.props.submit_button_text }
+						placeholder={ __( 'Submit »' ) }
+						onChange={ this.onChangeSubmit }
+					/>
+				</div>
+			</Fragment>
+		);
+	}
+}
+
+function GrunionFieldRequiredToggle( props ) {
+	return (
+		<ToggleControl
+			label={ __( 'Required' ) }
+			checked={ props.required }
+			onChange={ props.onChange }
+		/>
+	);
+}
+
+class GrunionFieldSettings extends Component {
+	constructor( ...args ) {
+		super( ...args );
+		this.onChangeRequired = this.onChangeRequired.bind( this );
+	}
+
+	onChangeRequired( x ) {
+		this.props.setAttributes( { required: x } );
+	}
+
+	render() {
+		return (
+			<InspectorControls>
+				<PanelBody title={ __( 'Field Settings', 'jetpack' ) }>
+					<GrunionFieldRequiredToggle
+						required={ this.props.required }
+						onChange={ this.onChangeRequired }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+	}
+}
+
+class GrunionFieldLabel extends Component {
+	constructor( ...args ) {
+		super( ...args );
+		this.onChangeLabel = this.onChangeLabel.bind( this );
+	}
+
+	onChangeLabel( x ) {
+		this.props.setAttributes( { label: x.target.value } );
+	}
+
+	render() {
+		return (
+			<Fragment>
+				<input
+					type='text'
+					value={ this.props.label }
+					className='grunion-field-label'
+					onChange={ this.onChangeLabel }
+				/>
+				{ this.props.required && <span className="required">{ __( '(required)' ) }</span> }
```

Comment:
Translation is missing `jetpack` textdomain.

Reply:
Yeah, I'm waiting until the end to do a full pass adding it in everywhere.  Also, I understand there's some confusion about what the proper way to handle i18n strings is anyways, so may change if that shakes up.

Reply:
Just added it in.